### PR TITLE
[Perf][Qwen] Inline zero padding kernel during MTP decodes (1% decode speedup)

### PR DESCRIPTION
**Pull request by benchislett**
## Purpose

Eliminate this little zero-er kernel from the Qwen3.x GDN decode layer execution.
image

It's small, only nets a 0.5-1% speedup E2E at BS1, but enables more comprehensive PDL overlapping between the QKVZ projection and the causal convolution (since there is no longer a kernel between them) -- which I may optimize further in a follow up PR.

To eliminate the zero I evaluated several strategies, but inlining into the GDN kernel itself seems the cleanest: a net speedup for all batch sizes (up to 1024) as long as padding is <25% of total token count. 

We still apply the zeroing when the specialized kernel is not invoked. So, the contract has not changed -- GDN still zeros the padding slots to avoid NaNs

## Test Plan

Unit tests are included.

## Test Result

Seems to be passing locally. Will ensure they run and pass in CI to be sure

### CODE REVIEW

File: tests/kernels/test_fused_gdn_post_conv.py [hunk 1]
```diff
@@ -276,7 +277,9 @@ def test_fused_gdn_decode_post_conv_mtp_head_ratios(
     state_dtype: torch.dtype,
     norm_dtype: torch.dtype,
     output_gate_activation: str,
+    padding_tokens: int,
 ) -> None:
+    """Live outputs match the reference and padding is zero, even after empty requests."""
```
Comment (benchislett) [inline tests/kernels/test_fused_gdn_post_conv.py]:
```suggestion
    """Live outputs match the reference and padding is zero."""
```

File: vllm/model_executor/layers/mamba/gdn/qwen_gdn_linear_attn.py [hunk 2]
```diff
@@ -918,7 +918,9 @@ def forward_cuda(
             and self.norm.weight.dtype in (torch.bfloat16, torch.float32)
         )
         if use_fused_gdn_decode:
-            core_attn_out = torch.zeros(
+            # Padding slots are guaranteed to be cleared to avoid NaN
+            # Either with .zero_() or inlined within the GDN kernel 
+            core_attn_out = torch.empty(
```
Comment (vadiklyutiy) [inline vllm/model_executor/layers/mamba/gdn/qwen_gdn_linear_attn.py:923]: I see the comment above but IMO use `empty` instead of `zeros` still a bit risky. There are (and might be added) additional backends and they may cause a NaN. Resolving a bug with NaN may be a really tricky.

**Review by vadiklyutiy — COMMENTED** [review]
- pre-commit is red
 - you mentioned E2E runs, could you add details
